Sources/SpeziQuestionnaire/TimedWalkView.swift

```diff
@@ -0,0 +1,91 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import FHIRQuestionnaires
+import ModelsR4
+import ResearchKit
+import ResearchKitOnFHIR
+import SwiftUI
+
+
+/// Renders a FHIR `Questionnaire`.
+public struct TimedWalkView: View {
+    @EnvironmentObject private var questionnaireDataSource: QuestionnaireDataSource
+    
+    @Binding private var isPresented: Bool
+    
+    private let identifier: String?
+    private let distanceInMeters: Double?
+    private let timeLimit: TimeInterval?
+    private let turnAroundTimeLimit: TimeInterval?
+    private let timedWalkResponse: ((ORKTimedWalkResult) async -> Void)?
+
+        
+    public var body: some View {
+        if let task = createTask() {
+            ORKOrderedTaskView(
+                tasks: task,
+                isPresented: $isPresented,
+                timedWalkResponse: { response in
+                    await timedWalkResponse?(response)
+                    await questionnaireDataSource.add(response)
+                },
+                tintColor: .accentColor
+            )
+                .ignoresSafeArea(.container, edges: .bottom)
+                .ignoresSafeArea(.keyboard, edges: .bottom)
+        } else {
+            Text("QUESTIONNAIRE_LOADING_ERROR_MESSAGE")
+        }
+    }
+    
+    
+    /// - Parameters:
+    ///   - questionnaire: The  `Questionnaire` that should be displayed.
+    ///   - isPresented: Indication from the questionnaire view if should be presented (not "Done" pressed or cancelled).
+    ///   - completionStepMessage: Optional completion message that can be appended at the end of the questionnaire.
+    ///   - questionnaireResponse: Optional response closure that can be used to manually obtain the `QuestionnaireResponse`.
+    public init(
+        identifier: String?,
+        distanceInMeters: Double?,
+        timeLimit: TimeInterval?,
+        turnAroundTimeLimit: TimeInterval?,
+        isPresented: Binding<Bool> = .constant(true),
+        timedWalkResponse: (@MainActor (ORKTimedWalkResult) async -> Void)? = nil
+    ) {
+        self.identifier = identifier
+        self.distanceInMeters = distanceInMeters
+        self.timeLimit = timeLimit
+        self.turnAroundTimeLimit = turnAroundTimeLimit
+        self._isPresented = isPresented
+        self.timedWalkResponse = timedWalkResponse
+    }
+    
+    
+    /// Creates a ResearchKit navigable task from a questionnaire
+    /// - Parameter questionnaire: a questionnaire
+    /// - Returns: a ResearchKit ordered task
+    private func createTask() -> ORKOrderedTask? {
```

Comment:
This function has questionnaire as an input for the questionnaireView. What should I have as an input here? Do I even need an input?

Reply:
@dguo8412 Discussed this today in our meeting, let me know if you have any follow up questions 👍

Reply:
I don't remember if we discussed this one in our meeting. Do we need an input for creating a task here? It seems to work fine without one.

Reply:
You are correct. You currently create the timed walk task right in the function. In the long run, you might have a `createActiveTask` function that takes the input and then creates a proper `ORK...` type but it is fine if this happens in the current function signature for now.